pkgs/development/coq-modules/coq-lsp/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, mkCoqDerivation, coq, serapi, version ? null }:
+
+mkCoqDerivation rec {
+  pname = "coq-lsp";
+  owner = "ejgallego";
+
+  useDune = true;
+
+  release."0.1.4+v8.16".sha256 = "sha256-5jztvAKpCFehtxu9Kw2inbPUZYVFIMcIoD5/u1WBuyI=";
+
+  inherit version;
+  defaultVersion = with lib.versions; lib.switch coq.coq-version [
+    { case = range "8.16" "8.17"; out = "0.1.4+v8.16"; }
+  ] null;
+
+  buildPhase = ''
+    runHook preBuild
+    dune build -p ${pname} -j $NIX_BUILD_CORES
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    dune install --prefix $out ${pname}
+    runHook postInstall
+  '';
+
```

Comment:
Isn’t that very close to the default values of `buildPhase` and `installPhase`? If possible I’d prefer to avoid redefining them.

Reply:
I updated the expression to use the default `buildPhase`, but I can't do the same with `installPhase` because it tries to install `$out/lib/coq` which is not present in coq-lsp
https://github.com/NixOS/nixpkgs/blob/be1950af0b7732ea7ef53634ef94a64956424ed3/pkgs/build-support/coq/default.nix#L119-L122